### PR TITLE
11.0 python2 sle

### PR DESCRIPTION
**Pull request by sle-odoo**

### CODE REVIEW

File: odoo/addons/base/ir/ir_qweb/qweb.py [hunk 1]
```diff
@@ -99,20 +99,35 @@ def visit_Lambda(self, node):
...
-            args=arguments(
-                args=args.args,
-                defaults=[self.visit(default) for default in args.defaults],
-                vararg=args.vararg,
-                kwarg=args.kwarg,
-                # assume we don't have any, not sure it's even possible to
-                # handle that cross-version
-                kwonlyargs=[],
-                kw_defaults=[],
-                posonlyargs=[],
```
Comment (xmo-odoo) [inline odoo/addons/base/ir/ir_qweb/qweb.py]: Fix the `arguments` shim instead (higher in the file). Either add `posonlyarg`, or replace `kwonlyargs, kw_defaults` by an ignored `**kwargs` at the end of the call.

File: odoo/addons/base/ir/ir_qweb/qweb.py [hunk 2]
```diff
@@ -547,17 +562,30 @@ def _create_def(self, options, body, prefix='fn', lineno=None):
...
+                name=name,
+                args=arguments(args=[
+                    arg(arg='self', annotation=None),
+                    arg(arg='append', annotation=None),
+                    arg(arg='values', annotation=None),
+                    arg(arg='options', annotation=None),
+                    arg(arg='log', annotation=None),
+                ], defaults=[], vararg=None, kwarg=None, posonlyargs=[], kwonlyargs=[], kw_defaults=[]),
+                body=body or [ast.Return()],
+                decorator_list=[])
```
Comment (xmo-odoo) [inline odoo/addons/base/ir/ir_qweb/qweb.py]: See above.

File: odoo/addons/base/ir/ir_qweb/qweb.py (hunk 1 above)
Comment (xmo-odoo) [reply odoo/addons/base/ir/ir_qweb/qweb.py]: Also you'll probably want to fw-bot ignore as the file was moved in 12.0 (and the shim was apparently removed in saas-12.1 / 13.0: https://github.com/odoo/odoo/commit/52f5528cfb6d1152ee226e268247dc685a8ba329#diff-56c5a32de746052fcf317821d868e243L38-L49)

File: odoo/addons/base/ir/ir_qweb/qweb.py [hunk 3]
```diff
@@ -555,7 +554,7 @@ def _create_def(self, options, body, prefix='fn', lineno=None):
                 arg(arg='values', annotation=None),
                 arg(arg='options', annotation=None),
                 arg(arg='log', annotation=None),
-            ], defaults=[], vararg=None, kwarg=None, posonlyargs=[], kwonlyargs=[], kw_defaults=[]),
+            ], defaults=[], vararg=None, kwarg=None, kwonlyargs=[], kw_defaults=[]),
```
Comment (xmo-odoo) [inline odoo/addons/base/ir/ir_qweb/qweb.py]: I'd think if @mart-e added the param it's necessary for 3.8 to work.

Comment (sle-odoo) [reply odoo/addons/base/ir/ir_qweb/qweb.py]: han ouais

Comment (mart-e) [reply odoo/addons/base/ir/ir_qweb/qweb.py]: Yep, at 04893ed868704d9cbf

Comment (sle-odoo) [reply odoo/addons/base/ir/ir_qweb/qweb.py]: laissez-moi le temps d'installer 3.8 :'(

**Comment by sle-odoo** [comment]
@mart-e if you could test it is working properly at your end it would be nice
i tested locally with 2.7.16 and 3.8.1, seems to be ok
thanks!

**Comment by mart-e** [comment]
@sle-odoo tested with success using
- 2.7.17
- 3.7.3
- 3.8.0

**Comment by xmo-odoo** [comment]
@robodoo r+

**Comment by sle-odoo** [comment]
@fw-bot ignore

**Comment by xmo-odoo** [comment]
@sle-odoo there are still 2 commits in the PR, shouldn't they be squashed?

**Comment by sle-odoo** [comment]
robodoo r+ fw-bot ignore

**Pull request closed without merging**